src/parser/deserializer.ts

```diff
@@ -286,14 +287,17 @@ function deserializeObject(
         (buffer[index++] << 8) |
         (buffer[index++] << 16) |
         (buffer[index++] << 24);
-    } else if (elementType === constants.BSON_DATA_NUMBER && promoteValues === false) {
-      dataView ??= new DataView(buffer.buffer, buffer.byteOffset, buffer.byteLength);
-      value = new Double(dataView.getFloat64(index, true));
-      index = index + 8;
     } else if (elementType === constants.BSON_DATA_NUMBER) {
-      dataView ??= new DataView(buffer.buffer, buffer.byteOffset, buffer.byteLength);
-      value = dataView.getFloat64(index, true);
-      index = index + 8;
+      FLOAT_WRITE_BYTES[0] = buffer[index++];
+      FLOAT_WRITE_BYTES[1] = buffer[index++];
+      FLOAT_WRITE_BYTES[2] = buffer[index++];
+      FLOAT_WRITE_BYTES[3] = buffer[index++];
+      FLOAT_WRITE_BYTES[4] = buffer[index++];
+      FLOAT_WRITE_BYTES[5] = buffer[index++];
+      FLOAT_WRITE_BYTES[6] = buffer[index++];
+      FLOAT_WRITE_BYTES[7] = buffer[index++];
```

Comment:
Let's change this to a `for` loop instead. 

```suggestion
     for (let i = 0; i < 8; i++) {
         FLOAT_WRITE_BYTES[i] = buffer[index++];
     }
```

Reply:
For very small amounts of code, writing it out proves to be more performant than writing a loop